lib/plug/static.ex

```diff
@@ -182,6 +186,100 @@ defmodule Plug.Static do
     h in only or match?({0, _}, prefix != [] and :binary.match(h, prefix))
   end
 
+  defp serve_range(conn, path, segments, [], %{gzip?: gzip?, brotli?: brotli?} = options) do
+    encoding = file_encoding(conn, path, gzip?, brotli?)
+    serve_static(encoding, segments, options)
+  end
+  defp serve_range(conn, path, segments, [range], options) do
+    encoding = file_encoding(conn, path, false, false)
+    serve_range(encoding, range, segments, options)
+  end
+  defp serve_range(_conn, _path, _segments, _range, _options) do
+    raise InvalidRangeError
+  end
+
+  @byte_range_pattern ~r/^\s*bytes=([0-9]+)?-([0-9]+)?\s*$/
+
+  defp serve_range({:ok, conn, file_info, path}, range, segments, options) do
+    file_size = elem(file_info, 1)
+
+    parsed_range =
+      @byte_range_pattern
+      |> Regex.run(range)
+      |> start_and_end(file_size)
+      |> check_bounds(file_size)
+
+    case parsed_range do
+      {range_start, range_end} ->
+        send_range(conn, path, range_start, range_end, file_size, segments, options)
+      :entire_file ->
+        serve_static({:ok, conn, file_info, path}, segments, options)
+      :invalid ->
+        raise InvalidRangeError
```

Comment:
Is raising the reasonable thing to do here? You said we don't support all syntaxes, so maybe we should rather send the whole file instead of crashing.